Verify number of InfluxDB series migrated

trevorbonas opened the pull request:
*Issue #, if available:* N/A

*Description of changes:*

- The series count of buckets to be backed up are now logged when log level is set to `debug`.
- The series count of buckets after restoration are now logged when log level is set to `debug`.
- `bucket_exists` has been improved.
- All logs now include a timestamp.
- Documentation describing how series counts are logged has been added.
- `scripts/temp` has been added to `.gitignore`.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

trevorbonas committed:
Verify number of series migrated

- The series count of buckets to be backed up are now logged when log level is set to `debug`.
- The series count of buckets after restoration up are now logged when log level is set to `debug`.
- `bucket_exists` has been improved.
- All logs now include a timestamp.
- Documentation describing how series counts are logged has been added.
- `scripts/temp` has been added to `.gitignore`.

trevorbonas committed:
Add verify_ssl to series count functions

sethusrinivasan commented on tools/python/influx-migration/influx_migration.py:
```diff
@@ -119,14 +119,15 @@ def report_all_bucket_series_count(host, token, org_name=None):
                 offset=offset)
 
-def report_bucket_series_count(bucket_name, host, token, org_name=None):
+def report_bucket_series_count(bucket_name, host, token, org_name=None, skip_verify=False):
```
Can the last parameter to this method be "verify_ssl" that you can pass in the value as-is to  InfluxDBClient()?

trevorbonas replied:
`skip_verify` originates from the CLI argument `--skip-verify`. This is also what the Influx CLI uses. The `--skip-verify` argument is passed essentially as-is to all places the Influx CLI is used.

This means that the value cannot be currently passed in as is, since the value will originate as `--skip-verify`, having the opposite meaning of `--verify-ssl`.

We could instead change the CLI option from `--skip-verify` to `--verify-ssl`, allowing the value to be passed as is. But then the value could not be passed to the Influx CLI as is, and would have to be negated.

rajeiyer commented on tools/python/influx-migration/influx_migration.py:
```diff
@@ -119,14 +119,15 @@ def report_all_bucket_series_count(host, token, org_name=None):
                 offset=offset)
 
-def report_bucket_series_count(bucket_name, host, token, org_name=None):
+def report_bucket_series_count(bucket_name, host, token, org_name=None, skip_verify=False):
```
This method is invoked to log the series count in a bucket, but it raises an exception if the count is not found. Why do we need to raise an exception here instead of logging it as an error?

trevorbonas replied:
If you're referring to the raised `ValueError` within the `try` block, this exception will be caught immediately. I am raising this exception instead of logging the error immediately because I am using everything under the `except` block as a place to log all failures related to being unable to log a bucket's series count, to prevent duplicated code.

trevorbonas committed:
Improve bucket count calls

Now all calls to report_bucket_series_count and
report_all_bucket_series_count include the skip_verify argument instead
of defaulting to False.

Calls to check bucket series counts after a full migration have been
improved:
- If a full migration without --csv has been
done then the destination instance will have the source operator token,
which will be used for checking all bucket series counts.
- If --csv is used with a full migration then tokens will not be migrated
and these calls will instead use the original destination token.

trevorbonas commented:
I have added some improvements for the bucket series count function calls.
- All calls now use the user's `--skip-verify` value, instead of leaving the argument out and defaulting to `False`.
- The lack of token migration has been taken into account for when users do a full migration with `--csv`.

alexey-temnikov commented on tools/python/influx-migration/influx_migration.py:
```diff
@@ -49,6 +50,11 @@
+# The number of seconds to wait before scraping from the /metrics endpoint
+METRICS_SCRAPE_INTERVAL_SECONDS=10
+
+BUCKET_PAGINATION_LIMIT=100
```
nit. add comment is that 100 is max allowed. or use constant MAX_BUCKET_PAGINATION_LIMIT
